examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -162,132 +162,141 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
     bool requestorCanConsent              = commandData.requestorCanConsent.ValueOr(false);
     QueryImageResponse::Type response;
 
-    switch (mQueryImageBehavior)
+    if (mIgnoreQueryImageCount == 0)
```

Comment:
To avoid more lines of change than necessary, you could do something like:

```
if (mIgnoreQueryImageCount > 0)
{
         ChipLogDetail(SoftwareUpdate, "Skip HandleQueryImage response. mIgnoreQueryImageCount %" PRIu32, mIgnoreQueryImageCount);
         mIgnoreQueryImageCount--;
         return EMBER_ZCL_STATUS_SUCCESS;
}
```